Docker Weather API

N1khto opened the pull request.

N1khto committed:
Docker Weather API

N1khto committed:
Added empty line at the end of dockerfile

ArsenPidhoretskyi commented on app/main.py:
```diff
@@ -1,6 +1,23 @@
-    # write your code here
-    pass
+    result = requests.get(URL + f"q={FILTERING}&key={API_KEY}")
+    result = json.loads(result.content)
```
You can do as follow:
result.json()

ArsenPidhoretskyi commented on app/main.py:
```diff
@@ -1,6 +1,23 @@
+    result = requests.get(URL + f"q={FILTERING}&key={API_KEY}")
+    result = json.loads(result.content)
+    if result.get("error"):
+        print(f"{result.get('error').get('message')}")
```
If .get('error') return none you will have error
Add default argument

ArsenPidhoretskyi commented on app/main.py:
```diff
@@ -1,6 +1,23 @@
+    if result.get("error"):
+        print(f"{result.get('error').get('message')}")
+        return
+    print(f"{result.get('location').get('tz_id')} "
+          f"{result.get('current').get('last_updated')} "
+          f"Weather: {result.get('current').get('temp_c')} Celsius",
+          f"{result.get('current').get('condition').get('text')}")
```
The same with .get
Use [] or add default

ArsenPidhoretskyi reviewed (changes requested):
.

N1khto committed:
Fixes:
- added `Performing request` message;
- added default value to error result message
- used [] in correct respond message

ArsenPidhoretskyi commented on app/main.py:
```diff
@@ -1,6 +1,22 @@
+    print(f"Performing request to Weather API for city {FILTERING}...")
+    result = (requests.get(URL + f"q={FILTERING}&key={API_KEY}")).json()
+    if result.get("error"):
+        print(f"{result.get('error').get('message', 'oops, something wrong')}")
```
```suggestion
        print(f"{result['error'].get('message', 'oops, something wrong')}")
```

N1khto committed:
Update app/main.py

ArsenPidhoretskyi reviewed (approved):
Good job!
